apps/small_filters/process.h

```diff
@@ -0,0 +1,526 @@
+#ifndef PROCESS_H
+#define PROCESS_H
+
+#include "HalideRuntimeHexagonHost.h"
+#include "HalideBuffer.h"
+
+#define DECL_WEAK_PIPELINE3S(x) int x##_cpu(buffer_t *, buffer_t *, buffer_t *) __attribute__((weak)); \
```

Comment:
We shouldn't be doing this, it's going to make it hard to use halide_buffer_t, buffer_t will be deprecated (? @abadams), we shouldn't be writing new code that explicitly depends on it.